shared/src/test/diff/mlscript/Under.mls

```diff
@@ -56,12 +56,12 @@ def foo { _ ; _ } = 1
 //│ ║  l.48: 	def foo { _ ; _ } = 1
 //│ ╙──      	              ^
 //│ foo: {_: anything, _: anything} -> 1
-//│    = 1
+//│    = undefined
```

Comment:
Why is `res` overridden here?

Reply:
Since I removed all `res = VariableName`, we are not able to get the value by querying `res` any longer. Instead, now I search `VariableName` in `globalThis`  directly (here `VariableName` is `foo`).  However, it generates no JavaScript code and that's why it is `undefined`.

Reply:
Do you know how come it generates no code? If the generation fails, it should at least print a code-gen error. Is am error swallowed somewhere in there?

Reply:
I did find some mistakes in my code. I will figure it out later. Please ignore my replies above.

Reply:
> Should it be __= _ ?

You mean a default parameter? Why?

Generating `{ __ : _ }` is the right thing: a record field named `__` locally being given name `_` (which is already defined).

Reply:
> It leads to Error: Duplicate parameter name not allowed in this context, and that's why globalThis.foo5 is undefined.

Ok, so this is a case of the usual problem that we don't currently catch syntax errors as there is no easy way of doing it. It's fine, but can you please add a comment that explains this is the reason we get `undefined`?